modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/RocksDbMvPartitionStorage.java

```diff
@@ -657,21 +684,25 @@ private ReadResult readResultFromKeyAndValue(boolean isWriteIntent, ByteBuffer k
      * @return Read result.
      */
     private ReadResult readByTimestamp(RocksIterator seekIterator, RowId rowId, HybridTimestamp timestamp) {
-        ByteBuffer keyBuf = prepareHeapKeyBuf(rowId);
+        ByteBuffer keyBuf = prepareDirectKeyBuf(rowId);
 
         // Put timestamp restriction according to N2O timestamps order.
         putTimestampDesc(keyBuf, timestamp);
 
+        keyBuf.rewind();
+
         // This seek will either find a key with timestamp that's less or equal than required value, or a different key whatsoever.
         // It is guaranteed by descending order of timestamps.
-        seekIterator.seek(keyBuf.array());
+        seekIterator.seek(keyBuf);
+
+        keyBuf.rewind();
 
         return handleReadByTimestampIterator(seekIterator, rowId, timestamp, keyBuf);
     }
 
     /**
-     * Walks "version chain" via the iterator to find a row by timestamp.
-     * See {@link MvPartitionStorage#read(RowId, HybridTimestamp)} for details.
+     * Walks "version chain" via the iterator to find a row by timestamp. See {@link MvPartitionStorage#read(RowId, HybridTimestamp)} for
```

Comment:
Why would you re-format perfectly fine-looking comment, that specifically showed different sentences on different lines? Please don't do that, and return the old formatting back.

Reply:
Because that's what IDEA does with its autoformatting. If you want to have two sentences on separate lines, you should use `<p>` tags

Reply:
As far as I know, most people read javadocs in code, without HTML rendering. I meant that.